### PR TITLE
profile support

### PR DESCRIPTION
**Pull request by gaelrehault**

### CODE REVIEW

File: src/deploy/osp_deployer/profile.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                                  "::" + set + \
+                                  " is not valid for the profile " + \
+                                  self.profile_name + \
+                                  " it should be to one of the following " + \
+                                  "options : " + \
+                                  json.dumps(allowed_settings)
+                    elif validated is True:
+                        pass
+                    else:
+                        usr_sett = != user_config.get(str(stanza), set)
```
Comment (cdearborn) [inline src/deploy/osp_deployer/profile.py]: Looks like a syntax error on the above line? "= !="

File: src/deploy/osp_deployer/settings_sanity.py [hunk 2]
```diff
@@ -50,7 +51,7 @@ def check_files(self):
         logger.debug("Check settings ip's are valid.")
         shouldbbevalidips = [
             'public_api_gateway',
-            'public_api_netmask', 'public_api_allocation_pool_start',
+            'public_api_netmask',
```
Comment (cdearborn) [inline src/deploy/osp_deployer/settings_sanity.py]: We still want public_api_allocation_pool_start in here don't we?

File: src/deploy/osp_deployer/settings/sample-fx.properties [hunk 3]
```diff
@@ -1,197 +0,0 @@
-[
```
Comment (cdearborn) [inline src/deploy/osp_deployer/settings/sample-fx.properties]: Would you mind moving the removal of this file into the PR for removing FX support?

File: src/deploy/osp_deployer/settings_sanity.py (hunk 2 above)
Comment (gaelrehault) [reply src/deploy/osp_deployer/settings_sanity.py]: That particular validation moved into the profiles json file so no longer required here.

File: src/deploy/osp_deployer/settings/sample-fx.properties (hunk 3 above)
Comment (gaelrehault) [reply src/deploy/osp_deployer/settings/sample-fx.properties]: done

File: src/deploy/osp_deployer/profile.py (hunk 1 above)
Comment (gaelrehault) [reply src/deploy/osp_deployer/profile.py]: thanks, fixed

**Comment by jtwill98** [comment]
+1

**Pull request merged**